icicle-compiler/test/Icicle/Test/Arbitrary/Source.hs

```diff
@@ -342,11 +342,17 @@ genExp tgi
 
   primApps
    = do p <- arbitrary
-        let ft = testFresh "" $ primLookup' p
-        let num = length $ functionArguments ft
-        xs <- vectorOf num (genExp tgi)
+        let ft  = testFresh "" $ primLookup' p
+        let num = lengthFunctionArguments ft
+        xs     <- vectorOf num (genExp tgi)
         return $ foldl (App ()) (Prim () p) xs
 
+  lengthFunctionArguments t
```

Comment:
Unfortunately this fix for the tests doesn't work completely.

Reply:
Hmm. Weird, looks good to me